dali/pipeline/operators/crop/crop_attr.h

```diff
@@ -36,43 +36,55 @@ class CropAttr {
   explicit inline CropAttr(const OpSpec &spec)
     : spec__(spec)
     , batch_size__(spec__.GetArgument<int>("batch_size")) {
-    int crop_h = 0, crop_w = 0;
+    int crop_h = 0, crop_w = 0, crop_d = 0;
     bool has_crop_arg = spec__.HasArgument("crop");
     bool has_crop_w_arg = spec__.ArgumentDefined("crop_w");
     bool has_crop_h_arg = spec__.ArgumentDefined("crop_h");
-    is_whole_image_ = !has_crop_arg && !has_crop_w_arg && !has_crop_h_arg;
+    bool has_crop_d_arg = spec__.ArgumentDefined("crop_d");
+    is_whole_image_ = !has_crop_arg && !has_crop_w_arg && !has_crop_h_arg && !has_crop_d_arg;
 
     DALI_ENFORCE(has_crop_w_arg == has_crop_h_arg,
       "`crop_w` and `crop_h` arguments must be provided together");
 
+    if (has_crop_d_arg) {
+      DALI_ENFORCE(has_crop_w_arg,
+        "`crop_d` argument must be provided together with `crop_w` and `crop_h`");
+    }
+
     if (has_crop_arg) {
-      DALI_ENFORCE(!has_crop_h_arg && !has_crop_w_arg,
-        "`crop` argument is not compatible with `crop_h`, `crop_w`");
+      DALI_ENFORCE(!has_crop_h_arg && !has_crop_w_arg && !has_crop_d_arg,
+        "`crop` argument is not compatible with `crop_h`, `crop_w`, `crop_d`");
 
       auto cropArg = spec.GetRepeatedArgument<float>("crop");
       DALI_ENFORCE(cropArg.size() > 0 && cropArg.size() <= 2);
       crop_h = static_cast<int>(cropArg[0]);
-      crop_w = static_cast<int>(cropArg.size() == 2 ? cropArg[1] : cropArg[0]);
+      crop_w = static_cast<int>(cropArg.size() >= 2 ? cropArg[1] : cropArg[0]);
```

Comment:
```
    if (cropArg.size() == 3) {
      crop_d = cropArg[0];
      crop_h = cropArg[1];
      crop_w = cropArg[2];
    } else {
       // do it the old way
    }
```

Reply:
I am changing this in the next step PR #1210